python/ray/_raylet.pyx

```diff
@@ -2799,7 +2799,7 @@ cdef class _GcsSubscriber:
         # subscriber_id needs to match the binary format of a random
         # SubscriberID / UniqueID, which is 28 (kUniqueIDSize) random bytes.
         subscriber_id = bytes(bytearray(random.getrandbits(8) for _ in range(28)))
-        gcs_address, gcs_port = address.split(":")
+        gcs_address, gcs_port = address.rsplit(":", 1)
```

Comment:
If `address` is `"[::1]:6379"`, `gcs_address` remains `"[::1]"`. The C++ subscriber likely expects a bare IP without brackets. I think you need to strip the brackets `[`/`]` before passing into the C++ API (or preferably use a helper that handles both cases, and the helper method can be unit tested).

Reply:
fixed